addons/website_sale/models/res_partner.py

```diff
@@ -27,7 +27,7 @@ def _onchange_property_product_pricelist(self):
                 ),
             }}
 
-    def _get_current_partner(self, order_sudo=False, **kwargs):
+    def _get_current_partner(self, *, order_sudo=False, **kwargs):
```

Comment:
I don't understand why this is needed, the base method doesn't have any args and I didn't found any other overrides

Reply:
Why do we even need this kind of linter? This method is working fine the way it is, I don't find this better in any sense

Reply:
It is making sure that we don't use `order_sudo` as a positional parameter in other modules.

For instance, in a module that depends on `portal` but not `website_sale`, if you used `order_sudo` as a positional parameter, it would work fine on runbot because all the modules are installed.
But then as soon as a customer installs that module without `website_sale` it would break.

This is a good test that would have prevented more than a few issues in the past, it doesn't cost much, and the change on this line makes sense.

Reply:
`order_sudo` shouldn't be used as a positional parameter, it would also break if any module that overrides this method is installed after `website_sale`, it doesn't make sense. 

I'd rather do this then:
```suggestion
    def _get_current_partner(self, **kwargs):
       order_sudo = kwargs.get('order_sudo', False)
```

Reply:
> `order_sudo` shouldn't be used as a positional parameter

They why define it as such?
It's exactly what the `*` does by defining it as a keyword-only argument

Sure, you can also use `kwargs.get`, but it's a bit heavier in the body of the function then. It's a tradeoff, personally I don't mind. My point was just that adding `*` in here is a good idea and is justified.

Reply:
If I understand the diff, `*` allows you to pass any argument without throwing an error. I expect errors when parameters are not sent correctly. From my POV, in this specific case, this makes room for more bad code rather than a clean one.

Reply:
> If I understand the diff, * allows you to pass any argument without throwing an error

This is not what it does. A single `*` marks a separation between positional-or-keyword parameters and keyword-only parameters.

This is not the same as `*x`.
```py
>>> def foo(*):
  File "<stdin>", line 1
    def foo(*):
            ^
SyntaxError: named arguments must follow bare *
```
```py
>>> def foo(*, a):
...     pass
... 
>>> foo(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() takes 0 positional arguments but 1 was given
>>> foo(b=2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() got an unexpected keyword argument 'b'
```

Reply:
https://peps.python.org/pep-3102/

Reply:
Thanks for the explanation. It's clearer in my mind now, I still don't like it but I see the added value